File: Obchisluvannya/threaddemosharp/threaddemo/ThreadCalculator.cs
using threaddemo;

public class ThreadCalculator
{
    private readonly Thread thread;
    private int id;
    public int ThreadId
    {
        get { return id; }
        set { id = value; }
    }

    private volatile bool run = true;

    public ThreadCalculator(int id)
    {
        thread = new Thread(Calculator);
        ThreadId = id;
    }

    public void Stop()
    {
        run = false;
    }

    private void Calculator()
    {
        double sum = 0;
        double count = 0;
        do
        {
            count++;
            sum += 3;
        } while (run);
        Console.WriteLine($"ThreadId: {ThreadId} | Sum: {sum} | Amount of elements: {count}");
    }

    public void Start()
    {
        thread.Start();
    }
}

File: Obchisluvannya/threaddemo/src/com/company/ThreadCalculator.java
package com.company;

public class ThreadCalculator implements Runnable {
    private final Thread thread;
    private int id;
    private volatile boolean running = true;

    public ThreadCalculator(int id) {
        this.id = id;
        this.thread = new Thread(this);
    }

    public int getThreadId() {
        return id;
    }

    public void stop() {
        running = false;
    }

    public void start() {
        thread.start();
    }

    @Override
    public void run() {
        double sum = 0;
        double count = 0;
        
        while (running) {
            count++;
            sum += 3;
        }
        
        System.out.printf("ThreadId: %d | Sum: %.0f | Amount of elements: %.0f%n", 
                id, sum, count);
    }
}